aws/protocol/restjson/decoder_util.go

```diff
@@ -2,68 +2,48 @@ package restjson
 
 import (
 	"encoding/json"
-	"fmt"
 	"io"
 	"strings"
 
 	"github.com/awslabs/smithy-go"
-	smithyjson "github.com/awslabs/smithy-go/json"
 )
 
 // GetErrorInfo util looks for code, __type, and message members in the
 // json body. These members are optionally available, and the function
 // returns the value of member if it is available. This function is useful to
 // identify the error code, msg in a REST JSON error response.
 func GetErrorInfo(decoder *json.Decoder) (errorType string, message string, err error) {
-	startToken, err := decoder.Token()
-	if err == io.EOF {
-		return "", "", nil
+	var errInfo struct {
+		Code    string `json:"code"`
+		Type  string `json:"__type"`
+		Message string `json:"message"`
 	}
```

Comment:
Are we positive that there are no `Code` (capital `C`) cases?  Might be better not to have struct tags for these members so that the decoder's caseless field matching will be used.

Reply:
Thanks for pointing it out. Fixed.